File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int maxWidth = 0;
        int maxHeight = 0;
        
        for (int[] size : sizes) {
            // 큰 값을 가로, 작은 값을 세로로 정렬
            int w = Math.max(size[0], size[1]);
            int h = Math.min(size[0], size[1]);
            
            // 최대 가로 길이와 최대 세로 길이를 갱신
            if (w > maxWidth) maxWidth = w;
            if (h > maxHeight) maxHeight = h;
        }
        
        // 지갑의 크기는 최대 가로 길이와 최대 세로 길이의 곱
        return maxWidth * maxHeight;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
from typing import List

def solution(sizes: List[List[int]]) -> int:
    max_width = 0
    max_height = 0

    for size in sizes:
        width = max(size)  # 명함의 가로 길이
        height = min(size)  # 명함의 세로 길이
        
        # 최대 가로 길이와 최대 세로 길이 갱신
        if width > max_width:
            max_width = width
        if height > max_height:
            max_height = height

    return max_width * max_height